src/algebraic_geometry/pullbacks.lean

```diff
@@ -0,0 +1,330 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import algebraic_geometry.gluing
+import category_theory.limits.opposites
+import algebraic_geometry.Gamma_Spec_adjunction
+
+/-!
+# Fibred products of schemes
+
+In this file we construct the fibred product of schemes via gluing.
+We roughly follow [har77] Theorem 3.3.
+
+In particular, the main construction is to show that for an open cover `{ Uᵢ }` of `X`, if there
+exist fibred products `Uᵢ ×[Z] Y` for each `i`, then there exists a fibred product `X ×[Z] Y`.
+
+Then, for constructing the fibred product for arbitrary schemes `X, Y, Z`, we can use the
+construction to reduce to the case where `X, Y, Z` are all affine, where the fibred products is
+constructed via tensor products.
+
+-/
+universes v u
+noncomputable theory
+
+open category_theory category_theory.limits algebraic_geometry
+namespace algebraic_geometry.Scheme
```

Comment:
```suggestion
namespace algebraic_geometry.Scheme.fibre_product_construction
```
or something like that. Because there are many names like `t`, `t'`, `p1` etc, that shouldn't simply end up in the `Scheme` namespace, I think.